common/persistence/cassandra/cassandraPersistence.go

```diff
@@ -1135,8 +1135,15 @@ func (d *cassandraPersistence) CreateWorkflowExecution(request *p.CreateWorkflow
 					}
 				}
 
-				msg := fmt.Sprintf("Workflow execution creation condition failed. WorkflowId: %v, columns: (%v)",
-					request.Execution.GetWorkflowId(), strings.Join(columns, ","))
+				if prevRunID, ok := previous["current_run_id"]; ok && prevRunID != request.Execution.GetRunId() {
```

Comment:
I believe you need to use
```
previous["current_run_id"].(gocql.UUID).String()
```
Otherwise it always compare gocql.UUID with string.

Reply:
Changed, Thanks for catching this